flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/MutationActionList.java

```diff
@@ -0,0 +1,371 @@
+		return this;
+	}
+
+	/**
+ 	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_FAMILY},
+	 * which will create an HBase {@link Delete} operation that deletes versions of the specified family with
+	 * timestamps less than or equal to the specified timestamp.
+	 *
+	 * @param family family name
+	 * @param timestamp version timestamp
+	 * @return this
+	 */
+	public MutationActionList addDeleteFamily(byte[] family, long timestamp) {
+		actions.add(new MutationAction(family, null, null, timestamp, 0, MutationAction.Type.DELETE_FAMILY));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_FAMILY},
+	 * which will create an HBase {@link Delete} operation that deletes all versions of the specified family.
+	 *
+	 * @param family family name
+	 * @return this
+	 */
+	public MutationActionList addDeleteFamily(byte[] family) {
+		actions.add(new MutationAction(family, null, null, Long.MAX_VALUE, 0, MutationAction.Type.DELETE_FAMILY));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_COLUMNS},
+	 * which will create an HBase {@link Delete} operation that deletes versions of the specified column with
+	 * timestamps less than or equal to the specified timestamp.
+	 *
+	 * @param family family name
+	 * @param qualifier column qualifier
+	 * @param timestamp version timestamp
+	 * @return this
+	 */
+	public MutationActionList addDeleteColumns(byte[] family, byte[] qualifier, long timestamp) {
+		actions.add(new MutationAction(family, qualifier, null, timestamp, 0, MutationAction.Type.DELETE_COLUMNS));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_COLUMNS},
+	 * which will create an HBase {@link Delete} operation that deletes all versions of the specified column.
+	 *
+	 * @param family family name
+	 * @param qualifier column qualifier
+	 * @return this
+	 */
+	public MutationActionList addDeleteColumns(byte[] family, byte[] qualifier) {
+		actions.add(new MutationAction(family, qualifier, null, Long.MAX_VALUE, 0, MutationAction.Type.DELETE_COLUMNS));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_COLUMN},
+	 * which will create an HBase {@link Delete} operation that deletes the specified version of the specified column.
+	 *
+	 * @param family family name
+	 * @param qualifier column qualifier
+	 * @param timestamp version timestamp
+	 * @return this
+	 */
+	public MutationActionList addDeleteColumn(byte[] family, byte[] qualifier, long timestamp) {
+		actions.add(new MutationAction(family, qualifier, null, timestamp, 0, MutationAction.Type.DELETE_COLUMN));
+		return this;
+	}
+
+	static class MutationAction {
```

Comment:
Will it not be better to expose MutationAction - and allow the user to create a MutationAction and then just add them to the MutationList. MutationAction can have all the different flavours of addXXX() APIs?  The return type of each will be a MutationAction. That will get added to a MutationActionList and so the getActions() in mutation Action will just return the actions added to it. What do you think? Related to the HBaseSink implementing RichSinkFunction I think people from Flink community can answer you the best.